dask/dataframe/groupby.py

```diff
@@ -289,8 +289,18 @@ def _nunique_df_chunk(df, *index, **kwargs):
     return grouped
 
 
+def _drop_duplicates_rename(df):
+    # Avoid duplicate index labels in a groupby().apply() context
+    # https://github.com/dask/dask/issues/3039
+    # https://github.com/pandas-dev/pandas/pull/18882
+    names = [None] * df.index.nlevels
+    result = df.drop_duplicates().rename_axis(names)
+    return result
```

Comment:
Nit: just return instead of assign -> return. Also, `rename_axis` copies the data by default, when I don't think we need to here.

```python
return df.drop_duplicates().rename_axis(names, copy=False)
```